kafka-rest/src/test/java/io/confluent/kafkarest/integration/AbstractConsumerTest.java

```diff
@@ -119,15 +134,25 @@ protected String startConsumeMessages(
             .post(Entity.entity(subscribeRequest, Versions.KAFKA_V2_JSON));
     assertEquals(Response.Status.NO_CONTENT.getStatusCode(), subscribeResponse.getStatus());
 
+    Response response = null;
     // Start consuming. Since production hasn't started yet, this is expected to timeout.
-    Response response =
-        request(instanceResponse.getBaseUri() + "/records").accept(expectedMediatype).get();
-
-    assertOKResponse(response, expectedMediatype);
+    try {
+      response =
+          request(instanceResponse.getBaseUri() + "/records").accept(expectedMediatype).get();
+      assertOKResponse(response, expectedMediatype);
+    } catch (AssertionError e) {
+      if (response != null && response.getStatus() == Response.Status.NOT_FOUND.getStatusCode()) {
+        pause();
+        response =
+            request(instanceResponse.getBaseUri() + "/records").accept(expectedMediatype).get();
+        assertOKResponse(response, Versions.KAFKA_V2_JSON);
```

Comment:
Let me first note that I don't currently understand how our Consume v2 weighted media types for responses work. But shouldn't the retry still expect the originally expected media type.

Reply:
No, oops, that's a copy paste error.  Good spot.  The test still passes though... I'll go find out why

Reply:
It worked anyway (even through the failure case when I forced the exception throw to test the error path) because I tested with ConsumerJsonTest...